ferry.go

```diff
@@ -344,7 +436,12 @@ func (f *Ferry) Start() error {
 
 	// TODO(pushrax): handle changes to schema during copying and clean this up.
 	f.BinlogStreamer.TableSchema = f.Tables
-	f.DataIterator.Tables = f.Tables.AsSlice()
+
+	// TODO: refactor the TableSchemaCache logic to remove this weird post initialize
+	//       set tables hack
+	if f.Verifier != nil {
+		f.Verifier.SetApplicableTableSchemaCache(f.Tables)
+	}
```

Comment:
f.Tables is only loaded in `Start`. I tried to move this to initialize, but I have to change the integration tests quite a bit as the integration tests depends the tables being loaded in Start.

I started this work but I took it out because it confused this PR too much. I'll create another PR to fix this after merging this one.